stdlib/src/sys/ffi.mojo

```diff
@@ -317,6 +317,47 @@ struct DLHandle(CollectionElement, CollectionElementNew, Boolable):
 
         return res
 
+    @always_inline
+    fn call[
+        name: StringLiteral,
+        return_type: AnyTrivialRegType = NoneType,
+        *T: AnyType,
+    ](self, *args: *T) -> return_type:
+        """Call a function with any amount of arguments.
+
+        Parameters:
+            name: The name of the function.
+            return_type: The return type of the function.
+            T: The types of `args`.
+
+        Args:
+            args: The arguments.
+
+        Returns:
+            The result.
+        """
+        return self.call[name, return_type](args)
+
+    fn call[
+        name: StringLiteral, return_type: AnyTrivialRegType = NoneType
+    ](self, args: VariadicPack[element_trait=AnyType]) -> return_type:
+        """Call a function with any amount of arguments.
+
+        Parameters:
+            name: The name of the function.
+            return_type: The return type of the function.
+
+        Args:
+            args: The arguments.
+
+        Returns:
+            The result.
+        """
+
+        debug_assert(self.check_symbol(name), "symbol not found: " + name)
+        var v = _LITRefPackHelper(args._value).get_loaded_kgen_pack()
+        return self.get_function[fn (__type_of(v)) -> return_type](name)(v)
```

Comment:
**Comment** Using `fn(__type_of(..)) -> ..` is a neat trick that makes it possible to "spell" the signature of a variadic function, nice job finding and using that here 🙂